aspnetcore/security/authentication/identity-configuration.md

```diff
@@ -2,22 +2,86 @@
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+[!code-csharp[Main](identity/sample/src/ASPNET-IdentityDemo-PrimaryKeysConfig/Startup.cs?range=58-65,84)]
+
+---
 
-Like the passwords policy, all the settings of the application's cookie can be changed in the startup class.
+`IdentityOptions.Password` has the following properties:
+* `RequireDigit`: Requires a number between 0-9 in the password. Defaults to true.
+* `RequiredLength`: The minimum length of the password. Defaults to 6.
+* `RequireNonAlphanumeric`: Requires a non-alphanumeric character in the password. Defaults to true.
+* `RequireUppercase`: Requires an upper case character in the password. Defaults to true.
+* `RequireLowercase`: Requires a lower case character in the password. Defaults to true.
+* `RequiredUniqueChars`: Requires the number of distinct characters in the password. Defaults to 1.
 
-[!code-csharp[Main](identity/sample/src/ASPET-IdentityDemo-PrimaryKeysConfig/Startup.cs?highlight=2&range=72-80)]
 
 ## User's lockout
 
-[!code-csharp[Main](identity/sample/src/ASPET-IdentityDemo-PrimaryKeysConfig/Startup.cs?highlight=2&range=67-70)]
+[!code-csharp[Main](identity/sample/src/ASPNETv2-IdentityDemo-Configuration/Startup.cs?range=29-30,39-42,50-52)]
+
+`IdentityOptions.Lockout` has the following properties:
+* `DefaultLockoutTimeSpan`: The amount of time a user is locked out when a lockout occurs. Defaults to 5 minutes.
+* `MaxFailedAccessAttempts`: The number of failed access attempts until a user is locked out, if lockout is enabled. Defaults to 5.
+* `AllowedForNewUsers`: Determines if a new user can be locked out. Defaults to true.
+
+
+## Sign in settings
+
+[!code-csharp[Main](identity/sample/src/ASPNETv2-IdentityDemo-Configuration/Startup.cs?range=29-30,44-46,50-52)]
+
+`IdentityOptions.SignIn` has the following properties:
+* `RequireConfirmedEmail`: Requires a confirmed email to sign in. Defaults to false.
+* `RequireConfirmedPhoneNumber`: Requires a confirmed phone number to sign in. Defaults to false.
+
+
+## User validation settings
+
+[!code-csharp[Main](identity/sample/src/ASPNETv2-IdentityDemo-Configuration/Startup.cs?range=29-30,48-52)]
+
+`IdentityOptions.User` has the following properties:
+* `RequireUniqueEmail`: Requires each User to have a unique email. Defaults to false.
+* `AllowedUserNameCharacters`: Allowed characters in the username. Defaults to abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._@+.
+
+## Application's cookie settings
+
+Like the passwords policy, all the settings of the application's cookie can be changed in the `Startup` class.
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+Under `ConfigureServices` in the `Startup` class, you can configure the application's cookie.
+
+[!code-csharp[Main](identity/sample/src/ASPNETv2-IdentityDemo-Configuration/Startup.cs?name=snippet_configurecookie)]
+
+`CookieAuthenticationOptions` has the following properties:
```

Comment:
The property descriptions should probably move outside of the tabs. When you click the 1.x tab, the descriptions don't display.

Reply:
Ah yeah I couldn't see that locally with the tabs.  The only problem is if I move it out, some of those properties don't exist in 1.x.  Do you want me to call out that this is 2+ only or just move it and call it good?